src/services/ipinfo-client.ts

```diff
@@ -0,0 +1,33 @@
+import { AxiosInstance, AxiosStatic } from 'axios';
+import { ResultWithoutExpectedErrors } from '../doppler-types';
+
+export interface IpinfoResult {
+  countryCode?: string | null;
+}
+
+export interface IpinfoClient {
+  getCountryCode(): Promise<ResultWithoutExpectedErrors<IpinfoResult>>;
+}
+
+export class HttpIpinfoClient implements IpinfoClient {
+  private readonly axios: AxiosInstance;
+
+  constructor({ axiosStatic }: { axiosStatic: AxiosStatic }) {
+    this.axios = axiosStatic.create();
+  }
+
+  public async getCountryCode(): Promise<ResultWithoutExpectedErrors<IpinfoResult>> {
+    const defaultCountryCode = 'AR';
+    try {
+      const response = await this.axios.request({
+        method: 'GET',
+        url: 'http://ipinfo.io/json',
+      });
+      console.log(response);
+      const countryCode = response.data && response.data.country ? response.data.country : 'AR';
+      return { success: true, value: { countryCode: countryCode } };
+    } catch (error) {
+      return { success: false, value: { countryCode: defaultCountryCode } };
+    }
```

Comment:
Why not simply return `AR` if something goes wrong?

Reply:
I don't understand... I have to return always an object like that... 🤔